j5/backends/hardware/sr/v4/serial/power_board.py

```diff
@@ -0,0 +1,273 @@
+        """
+        if identifier in range(6):
+            response = self.request_with_response(f"OUT:{identifier}:I?")
+            return float(response) / 1000
+        else:
+            raise ValueError(f"{identifier!r} is not a valid power output identifier")
+
+    def buzz(
+        self,
+        identifier: int,
+        duration: timedelta,
+        frequency: float,
+        blocking: bool,
+    ) -> None:
+        """
+        Queue a pitch to be played.
+
+        :param identifier: piezo identifier to play pitch on.
+        :param duration: duration of the tone.
+        :param frequency: Pitch of the tone in Hz.
+        :param blocking: whether the code waits for the buzz
+        :raises ValueError: invalid value for parameter.
+        :raises NotSupportedByHardwareError: unsupported pitch freq or length.
+        """
+        if identifier != 0:
+            raise ValueError(f"Invalid piezo identifier {identifier!r}; "
+                             f"the only valid identifier is 0.")
+
+        duration_ms = round(duration / timedelta(milliseconds=1))
+        max_duration = (2 ** 32) - 1  # int32 max
+        if duration_ms > max_duration:
+            raise NotSupportedByHardwareError(
+                f"Maximum piezo duration is {max_duration}ms.")
+        if duration_ms < 0:
+            raise ValueError("Duration must be positive.")
+
+        frequency_int = int(round(frequency))
+        if frequency_int > 10000:
+            raise NotSupportedByHardwareError("Maximum piezo frequency is 10kHz.")
+        if frequency_int < 0:
+            raise ValueError("Frequency must be positive.")
+
+        self.request(f"NOTE:{frequency_int}:{duration_ms}")
+
+        if blocking:
+            sleep(duration.total_seconds())
+
+    def get_button_state(self, identifier: int) -> bool:
+        """
+        Get the state of a button.
+
+        :param identifier: Button identifier to fetch state of.
+        :returns: state of the button.
+        :raises CommunicationError: Invalid response received.
+        :raises ValueError: invalid button identifier.
+        """
+        if identifier != 0:
+            raise ValueError(f"Invalid button identifier {identifier!r}; "
+                             f"the only valid identifier is 0.")
+        response = self.request_with_response("BTN:START:GET?")
+        internal_button, external_button = response.split(":", 1)
+        if internal_button == "1" or external_button == "1":
+            return True
+        elif internal_button == "0" or external_button == "0":
+            return False
+        else:
+            raise CommunicationError(f"Invalid response received: {response}")
+
+    def wait_until_button_pressed(self, identifier: int) -> None:
+        """
+        Halt the program until this button is pushed.
+
+        :param identifier: Button identifier to wait for.
+        """
+        while not self.get_button_state(0):
```

Comment:
Since the start button's state is latching you want to discard the first response value otherwise this returns immediately if the start button has been pressed at any point since it was last checked.

Reply:
Didn't initially understand what you meant by "latched" but that makes sense.